### PR TITLE
Add Elasticsearch helper binaries to release

### LINKED ISSUE #5499: [Feature]: Add Elasticsearch helper binaries to GitHub Releases
### Requirement

We automate installation of Jaeger with Ansible using the Linux AMD64 binaries. We use the es-index-cleaner and es-rollover binaries, but we have to build them ourselves since they're not included in the repo. Is it possible to include those binaries the release packages?

### Problem

The Jaeger helper tools for Elasticsearch are only provided as container images, not as binaries to download.

### Proposal

A couple ways of accomplishing this are:
- Include the built Elasticsearch tools in the tarfiles (would increase size of each tarfile)
- Add a separate set of tarfiles for the helper tools (would increase the number of tarfiles, signatures, SBOMs in the release list etc.)

### Open questions

_No response_

### PR DESCRIPTION
**Pull request by FlamingSaint**
## Which problem is this PR solving?
- Resolves #5499 
## Description of the changes
- Add the es helpers to scripts/package-deploy.sh

## How was this change tested?
- 

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by yurishkuro** [comment]
this workflow does not run in the CI, so how was this tested?

**Comment by FlamingSaint** [comment]
I am not sure how to test this. The only solution I can come up with is to make a test release and see if it those es-helpers are included in the tar (Right now they would be included in the same tar). Any better ways to test this ?

**Comment by yurishkuro** [comment]
run `scripts/package-deploy.sh` script with redirecting logs to a file and upload the log file here

**Comment by FlamingSaint** [comment]
The log file is same before the changes in this PR. 
To redirect logs : `exec > logfile.log 2>&1`
[logfile.log](https://github.com/jaegertracing/jaeger/files/15503147/logfile.log)

**Comment by FlamingSaint** [comment]
How do I proceed to test this ?

**Comment by yurishkuro** [comment]
I think we introduced a bug recently to the Makefile, change it to
```diff
diff --git a/Makefile b/Makefile
index 7055dad4..ff5440b5 100644
--- a/Makefile
+++ b/Makefile
@@ -95,7 +95,7 @@ test-and-lint: test fmt lint

 .PHONY: echo-version
 echo-version:
-       @echo "GIT_CLOSEST_TAG=$(GIT_CLOSEST_TAG)"
+       @echo "$(GIT_CLOSEST_TAG)"

 .PHONY: echo-all-pkgs
 echo-all-pkgs:
```
and you will need to run `make build-all-platforms` before the packaging script.

**Comment by FlamingSaint** [comment]
There seems to be an error when I run `make build-all-platforms` .

```bash
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
GOOS=linux GOARCH=amd64 make _build-platform-binaries
make[1]: Entering directory '/home/raghuram/OpenSource/jaeger'
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath   -o ./cmd/agent/agent-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/agent
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath  -tags ui -o ./cmd/all-in-one/all-in-one-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/all-in-one
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath   -o ./cmd/collector/collector-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/collector
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath  -tags ui -o ./cmd/query/query-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/query
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath   -o ./cmd/ingester/ingester-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/ingester
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath  -tags ui -o ./cmd/jaeger/jaeger-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/jaeger
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath   -o ./cmd/remote-storage/remote-storage-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/remote-storage
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -o ./examples/hotrod/hotrod-linux-amd64 ./examples/hotrod/main.go
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" -o ./cmd/tracegen/tracegen-linux-amd64 ./cmd/tracegen/
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" -o ./cmd/anonymizer/anonymizer-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/anonymizer/
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -o ./plugin/storage/es/esmapping-generator-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/esmapping-generator/
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -o ./cmd/es-index-cleaner/es-index-cleaner-linux-amd64 ./cmd/es-index-cleaner/
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -o ./cmd/es-rollover/es-rollover-linux-amd64 ./cmd/es-rollover/
make _build-platform-binaries-debug GOOS=linux GOARCH=amd64 DEBUG_BINARY=1
make[2]: Entering directory '/home/raghuram/OpenSource/jaeger'
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l"  -o ./cmd/agent/agent-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/agent
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l"  -o ./cmd/collector/collector-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/collector
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l" -tags ui -o ./cmd/query/query-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/query
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l"  -o ./cmd/ingester/ingester-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/ingester
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l"  -o ./cmd/remote-storage/remote-storage-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/remote-storage
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l" -tags ui -o ./cmd/all-in-one/all-in-one-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/all-in-one
make[2]: Leaving directory '/home/raghuram/OpenSource/jaeger'
make[1]: Leaving directory '/home/raghuram/OpenSource/jaeger'
make _prepare-winres-helper NAME="Jaeger Agent"            PKGPATH="cmd/agent"
make[1]: Entering directory '/home/raghuram/OpenSource/jaeger'
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
echo $VERSIONINFO | goversioninfo -o="cmd/agent/resource.syso" -
2024/05/31 05:02:21 Could not parse the .json file: invalid character ',' looking for beginning of value
make[1]: *** [Makefile:356: _prepare-winres-helper] Error 2
make[1]: Leaving directory '/home/raghuram/OpenSource/jaeger'
make: *** [Makefile:342: _prepare-winres] Error 2
```

**Comment by yurishkuro** [comment]
What does this command output for you?  `git rev-parse --is-shallow-repository`
If it prints true, can you make your repo non-shallow?

**Comment by FlamingSaint** [comment]
Screenshot from 2024-05-31 11-49-13

**Comment by yurishkuro** [comment]
That's odd, my understanding `fatal: No names found, cannot describe anything.` was logged by git when it's a shallow copy.

**Comment by yurishkuro** [comment]
so `make echo-version` doesn't print anything for you?

**Comment by FlamingSaint** [comment]
Yes,
image

Well I thought I might have some issues with GOPATH so I reinstalled go again. Looks like when I run `make build-all-platforms` there is a different error now.
```bash
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
GOOS=linux GOARCH=amd64 make _build-platform-binaries
make[1]: Entering directory '/home/raghuram/OpenSource/jaeger'
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath   -o ./cmd/agent/agent-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/agent
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath  -tags ui -o ./cmd/all-in-one/all-in-one-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/all-in-one
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath   -o ./cmd/collector/collector-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/collector
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath  -tags ui -o ./cmd/query/query-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/query
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath   -o ./cmd/ingester/ingester-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/ingester
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath  -tags ui -o ./cmd/jaeger/jaeger-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/jaeger
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath   -o ./cmd/remote-storage/remote-storage-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/remote-storage
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -o ./examples/hotrod/hotrod-linux-amd64 ./examples/hotrod/main.go
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" -o ./cmd/tracegen/tracegen-linux-amd64 ./cmd/tracegen/
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" -o ./cmd/anonymizer/anonymizer-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/anonymizer/
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -o ./plugin/storage/es/esmapping-generator-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/esmapping-generator/
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -o ./cmd/es-index-cleaner/es-index-cleaner-linux-amd64 ./cmd/es-index-cleaner/
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -o ./cmd/es-rollover/es-rollover-linux-amd64 ./cmd/es-rollover/
make _build-platform-binaries-debug GOOS=linux GOARCH=amd64 DEBUG_BINARY=1
make[2]: Entering directory '/home/raghuram/OpenSource/jaeger'
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l"  -o ./cmd/agent/agent-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/agent
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l"  -o ./cmd/collector/collector-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/collector
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l" -tags ui -o ./cmd/query/query-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/query
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l"  -o ./cmd/ingester/ingester-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/ingester
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l"  -o ./cmd/remote-storage/remote-storage-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/remote-storage
fatal: No names found, cannot describe anything.
CGO_ENABLED=0 installsuffix=cgo go build -trimpath -gcflags="all=-N -l" -tags ui -o ./cmd/all-in-one/all-in-one-debug-linux-amd64 -ldflags "-X github.com/jaegertracing/jaeger/pkg/version.commitSHA=665385b6975b0506ab2be0fcd77a7e48d8b3300e -X github.com/jaegertracing/jaeger/pkg/version.latestVersion= -X github.com/jaegertracing/jaeger/pkg/version.date=2024-05-30T22:29:36Z" ./cmd/all-in-one
make[2]: Leaving directory '/home/raghuram/OpenSource/jaeger'
make[1]: Leaving directory '/home/raghuram/OpenSource/jaeger'
make _prepare-winres-helper NAME="Jaeger Agent"            PKGPATH="cmd/agent"
make[1]: Entering directory '/home/raghuram/OpenSource/jaeger'
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
echo $VERSIONINFO | goversioninfo -o="cmd/agent/resource.syso" -
/bin/bash: line 1: goversioninfo: command not found
make[1]: *** [Makefile:356: _prepare-winres-helper] Error 127
make[1]: Leaving directory '/home/raghuram/OpenSource/jaeger'
make: *** [Makefile:342: _prepare-winres] Error 2

```

**Comment by FlamingSaint** [comment]
Yes, Apart form the usual fatal messages 
image.

**Comment by FlamingSaint** [comment]
I made a change in the Makefile :
```bash
GIT_SHALLOW_CLONE := $(shell git rev-parse --is-shallow-repository)
# Some of GitHub Actions workflows do a shallow checkout without tags. This avoids logging warnings from git.
GIT_CLOSEST_TAG=$(shell if [ "$(GIT_SHALLOW_CLONE)" = "false" ]; then git describe --abbrev=0 --tags; else echo 0.0.0; fi)
```
to this:
```bash 
"$(GIT_SHALLOW_CLONE)" = "true"
```


After this change the fatal messages stopped. This is very weird considering  `git rev-parse --is-shallow-repository` is false.

**Comment by FlamingSaint** [comment]
```bash
GIT_SHALLOW_CLONE := $(shell git rev-parse --is-shallow-repository)
# Some of GitHub Actions workflows do a shallow checkout without tags. This avoids logging warnings from git.
GIT_CLOSEST_TAG=$(shell if [ "$(GIT_SHALLOW_CLONE)" = "true" ]; then git describe --abbrev=0 --tags; else echo 0.0.0; fi)
```
I changed the value of `GIT_SHALLOW_CLONE` to true and the fatal errors where not there anymore. This is very weird considering that `git rev-parse --is-shallow-repository` is false.

File: scripts/package-deploy.sh [hunk 1]
```diff
@@ -9,12 +9,24 @@ function stage-platform-files {
...
-    cp "./cmd/query/query-${PLATFORM}"            "${PACKAGE_STAGING_DIR}/jaeger-query${FILE_EXTENSION}"
-    cp "./cmd/collector/collector-${PLATFORM}"    "${PACKAGE_STAGING_DIR}/jaeger-collector${FILE_EXTENSION}"
-    cp "./cmd/ingester/ingester-${PLATFORM}"      "${PACKAGE_STAGING_DIR}/jaeger-ingester${FILE_EXTENSION}"
-    cp "./examples/hotrod/hotrod-${PLATFORM}"     "${PACKAGE_STAGING_DIR}/example-hotrod${FILE_EXTENSION}"
+    cp "./cmd/all-in-one/all-in-one-${PLATFORM}"                    "${PACKAGE_STAGING_DIR}/jaeger-all-in-one${FILE_EXTENSION}"
+    cp "./cmd/agent/agent-${PLATFORM}"                              "${PACKAGE_STAGING_DIR}/jaeger-agent${FILE_EXTENSION}"
+    cp "./cmd/query/query-${PLATFORM}"                              "${PACKAGE_STAGING_DIR}/jaeger-query${FILE_EXTENSION}"
+    cp "./cmd/collector/collector-${PLATFORM}"                      "${PACKAGE_STAGING_DIR}/jaeger-collector${FILE_EXTENSION}"
+    cp "./cmd/ingester/ingester-${PLATFORM}"                        "${PACKAGE_STAGING_DIR}/jaeger-ingester${FILE_EXTENSION}"
+    cp "./examples/hotrod/hotrod-${PLATFORM}"                       "${PACKAGE_STAGING_DIR}/example-hotrod${FILE_EXTENSION}"
```
Comment (yurishkuro) [inline scripts/package-deploy.sh:17]: please restore to original indentation since you don't need this change

File: scripts/package-deploy.sh [hunk 2]
```diff
@@ -24,29 +36,45 @@ function package {
     local -r PLATFORM=$2
     local -r FILE_EXTENSION=${3:-}
     local -r PACKAGE_NAME=jaeger-$VERSION-$PLATFORM
+    local -r TOOLS_PACKAGE_NAME=jaeger-tools-$VERSION-$PLATFORM
     local -r PACKAGE_STAGING_DIR=$PACKAGE_NAME
```
Comment (yurishkuro) [inline scripts/package-deploy.sh:40]:
```suggestion
    local -r PACKAGE_STAGING_DIR=$PACKAGE_NAME
    local -r TOOLS_PACKAGE_NAME=jaeger-tools-$VERSION-$PLATFORM
```

**Comment by FlamingSaint** [comment]
> I think we introduced a bug recently to the Makefile, change it to
> 
> ```diff
> diff --git a/Makefile b/Makefile
> index 7055dad4..ff5440b5 100644
> --- a/Makefile
> +++ b/Makefile
> @@ -95,7 +95,7 @@ test-and-lint: test fmt lint
> 
>  .PHONY: echo-version
>  echo-version:
> -       @echo "GIT_CLOSEST_TAG=$(GIT_CLOSEST_TAG)"
> +       @echo "$(GIT_CLOSEST_TAG)"
> 
>  .PHONY: echo-all-pkgs
>  echo-all-pkgs:
> ```
> 
> and you will need to run `make build-all-platforms` before the packaging script.

Shall I include this change as well ?

**Comment by yurishkuro** [comment]
yes

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Thanks

**Pull request merged**